crates/primitives/src/account.rs

```diff
@@ -31,10 +36,97 @@ impl Account {
     }
 }
 
+/// Bytecode for an account.
+///
+/// A wrapper around [`revm::primitives::Bytecode`][RevmBytecode] with encoding/decoding support.
+///
+/// Note: Upon decoding bytecode from the database, you *should* set the code hash using
+/// [`Self::with_code_hash`].
+#[derive(Debug, Clone, Default, PartialEq, Eq, Serialize, Deserialize)]
+pub struct Bytecode(pub RevmBytecode);
```

Comment:
do we need this wrapper type?
we could also implement Compact via feature "revm" in codecs.

But perhaps wrapping via this new type is better.

Reply:
If we do that, we would need to re-export the type from reth-codecs in order to use it anywhere, seems like a weird structure imo. Generally the crates around db and codecs are a bit iffy to work with, especially since some of the impl of codecs is in reth-db, some is in reth-codecs, some is in reth-codecs-derive...

Defining it here makes it clearer to me imo